services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/domain/UserContact.kt

```diff
@@ -1,67 +1,30 @@
 package fi.hel.haitaton.hanke.domain
 
-import fi.hel.haitaton.hanke.application.ApplicationContactType
 import fi.hel.haitaton.hanke.application.ApplicationData
 import fi.hel.haitaton.hanke.application.CableReportApplicationData
 import fi.hel.haitaton.hanke.application.CustomerWithContacts
 
-sealed interface UserContact {
-    val name: String
-    val email: String
-}
-
-data class HankeUserContact(override val name: String, override val email: String) : UserContact {
-    companion object {
-        fun from(name: String?, email: String?): HankeUserContact? =
-            when {
-                name.isNullOrBlank() || email.isNullOrBlank() -> null
-                else -> HankeUserContact(name, email)
-            }
-    }
-}
-
-data class ApplicationUserContact(
-    override val name: String,
-    override val email: String,
-    val type: ApplicationContactType
-) : UserContact {
+data class UserContact(val name: String, val email: String) {
```

Comment:
This is only used for sending emails, so this might be better in the email package. The name is not very specific, so the package would be a nice hint as to what this is.

I've run across this class a couple of times now, and I have always had to remind myself what this is used for.

A KDoc﻿ comment explaining what this is wouldn't hurt either.

Reply:
This is not directly used in emails.

This is only(?) used in HankeKayttajaService to create tokens for application contacts / hanke yhteyshenkilo. The companion object provides a unified way to gather contacts that have a name and email address.

This being said, this should probably be located in permissions (same as kayttajaservice) package and the extension functions need to be moved somewhere else also.